ir/type.def

```diff
@@ -268,6 +269,7 @@ class Type_InfInt : Type, ITypeVar {
         return true; /* ignore declid */
     }
     const Type* getP4Type() const override { return this; }
+    int width_bits() const override { return 0; }
```

Comment:
Hmm, is this a good idea? `with_bits` fails for some types which don't have defined width, yet here it returns 0 as a "don't know" value (even though 0 can also be a perfectly valid with of `bit<0>`).

Reply:
Yes, it would be better to remove this, but there are many places in that seem to depend on this.

Reply:
Sadly, there are more cases that someone could have relied on (whether deliberately or as a bug, e.g. `void` and unknown). Ultimately I think the `BUG` makes sense though (and to me it would make sense for `int` too).

Reply:
A common case that hits this is code that is converting constants to strings of some kind (eg, json or other output).  It generally just calls `width_bits()` on the constant type and if it returns 0 ouputs a constant with no specific size (if >0, it outputs some size indications).  This happens to be what you want, but relies on InfInt::width_bits() returning 0.